src/__tests__/paragraph.test.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * Internal dependencies
+ */
+import Paragraph from '../../gutenberg/packages/block-library/src/paragraph/edit.native.js';
+
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+jest.mock( '@wordpress/blocks' );
+
+const onReplace = jest.fn();
+const insertBlocksAfter = jest.fn();
+const setAttributes = jest.fn();
+
+const getTestComponentWithContent = ( content ) => {
+	return shallow(
+		<Paragraph
+			attributes={ { content } }
+			setAttributes={ setAttributes }
+			onReplace={ onReplace }
+			insertBlocksAfter={ insertBlocksAfter }
+		/>
+	);
+};
+
+const getTestInstanceWithContent = ( content ) => {
+	return getTestComponentWithContent( content ).instance();
+};
+
+describe( 'Paragraph block', () => {
+	it( 'renders without crashing', () => {
+		const component = getTestComponentWithContent( '' );
+		expect( component.exists() ).toBe( true );
+	} );
+
+	it( 'split empty block', () => {
+		// Given
+		const instance = getTestInstanceWithContent( '' );
+
+		const blocks = [ {} ];
+		const before = null;
+		const after = null;
+
+		//When
+		instance.splitBlock( before, after, ...blocks );
+
+		//Then
+
+		// The block is deleted if before is null
+		expect( onReplace ).toHaveBeenCalledTimes( 1 );
```

Comment:
Interesting, I didn't know jest would reset its mock functions on each tests!
I think it's more common to declare those inside our tests to avoid side effects.
```js
const onReplace = jest.fn();
const insertBlocksAfter = jest.fn();
const setAttributes = jest.fn();
const instance = getTestInstanceWithContent( '', { onReplace, insertBlocksAfter } );
...
expect( onReplace ).toHaveBeenCalledTimes( 1 );
```

Reply:
This is a setting:
https://github.com/wordpress-mobile/gutenberg-mobile/blob/d91cd0459eb0e31e314f504a517b90c0253ac3a5/jest.config.js#L17

I thought it would be safer to have them automatically cleared to avoid unexpected side effects.
Do you prefer to do it manually to make the intention clear?

Reply:
Oh I see. I still think it's a little bit better to redefine those in each tests. Having self contained tests that only depends on the functions (including classes) we're testing is cleaner imo, it's closer to the functional programming paradigm.

Though it's very minor and I don't feel very strong about it, take it as my personal opinion :)

Reply:
Cool! I removed the global function definitions and now they are recreated on each test case.
I usually prefer to minimize the setup code on the case itself by creating helpers, but good call on avoid globals 👍 

I did keep the setting just in case :)